perf/time_series_updater.go

```diff
@@ -0,0 +1,133 @@
+package perf
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/evergreen-ci/utility"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/message"
+	"github.com/pkg/errors"
+)
+
+type ArgumentsModel struct {
+	Name  string      `json:"name"`
+	Value interface{} `json:"value"`
+}
+
+type TimeSeriesDataModel struct {
+	PerformanceResultID string  `json:"cedar_perf_result_id"`
+	Order               int     `json:"order"`
+	Value               float64 `json:"value"`
+	Version             string  `json:"version"`
+}
+
+type TimeSeriesModel struct {
+	Project     string                `json:"project"`
+	Variant     string                `json:"variant"`
+	Task        string                `json:"task"`
+	Test        string                `json:"test"`
+	Measurement string                `json:"measurement"`
+	Arguments   []ArgumentsModel      `json:"args"`
+	Data        []TimeSeriesDataModel `json:"data"`
+}
+
+type PerformanceAnalysisService interface {
+	ReportUpdatedTimeSeries(context.Context, TimeSeriesModel) error
+}
+
+type performanceAnalysisAndTriageClient struct {
+	user    string
+	token   string
+	baseURL string
+}
+
+func NewPerformanceAnalysisService(baseURL, user string, token string) PerformanceAnalysisService {
+	return &performanceAnalysisAndTriageClient{user: user, token: token, baseURL: baseURL}
+}
+
+func (spc *performanceAnalysisAndTriageClient) ReportUpdatedTimeSeries(ctx context.Context, series TimeSeriesModel) error {
+	startAt := time.Now()
+
+	if err := spc.doRequest(http.MethodPost, spc.baseURL+"/time_series/update", ctx, series); err != nil {
+		return errors.WithStack(err)
+	}
+
+	grip.Debug(message.Fields{
+		"message":       "Reported updated time series to performance analysis and triage service",
+		"update":        series,
+		"duration_secs": time.Since(startAt).Seconds(),
+	})
+
+	return nil
+}
+
+func (spc *performanceAnalysisAndTriageClient) doRequest(method, route string, ctx context.Context, in interface{}) error {
+	body, err := json.Marshal(in)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	conf := utility.HTTPRetryConfiguration{
```

Comment:
I would use the default HTTP config, and just add/change the things you want to configure.

Reply:
Most of this file is the same as `perf/change_point_detector.go` was. For whatever reason, my `git mv` didn't take so it looks like all the new code. That part is the same as the old though. I'm a little hesitant to change the config just because we had issues with networking in the past and seems to be working well now.

Reply:
sounds fair